packages/terra-list/src/List.jsx

```diff
@@ -243,12 +253,21 @@ const List = ({
     }
   };
 
-  const cloneListItem = (ListItem, provider) => React.cloneElement(ListItem, {
+  const getStyleforDrag = (ListItem, snapshot, provider) => {
+    const styleProperties = provider?.draggableProps?.style;
+    if (styleProperties && snapshot && snapshot.isDragging) {
+      styleProperties['z-index'] = ListItem?.props?.zIndex ? `${ListItem.props.zIndex}` : '6001';
```

Comment:
```suggestion
      styleProperties['z-index'] = ListItem.props.zIndex;
```
zIndex has got a default value so I think we don't need this check here

Reply:
HI @supreethmr , yeah i have checked for the default prop but it was not avaiable.
Thanks
<img width="929" alt="Screenshot 2023-12-06 at 7 50 44 PM" src="https://github.com/cerner/terra-core/assets/132886427/0f825236-a98b-49d3-9de3-937de4ab2a8b">